src/platforms/linux/linuxnetworkwatcher.cpp

```diff
@@ -0,0 +1,167 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "linuxnetworkwatcher.h"
+#include "leakdetector.h"
+#include "logger.h"
+#include "timersingleshot.h"
+
+#include <QtDBus/QtDBus>
+
+// https://developer.gnome.org/NetworkManager/stable/nm-dbus-types.html#NMDeviceType
+#ifndef NM_DEVICE_TYPE_WIFI
+#  define NM_DEVICE_TYPE_WIFI 2
+#endif
+
+// https://developer.gnome.org/NetworkManager/stable/nm-dbus-types.html#NM80211ApFlags
+#ifndef NM_802_11_AP_SEC_NONE
+#  define NM_802_11_AP_SEC_NONE 0x00000000
+#endif
+
+#ifndef NM_802_11_AP_SEC_PAIR_WEP40
+#  define NM_802_11_AP_SEC_PAIR_WEP40 0x00000001
+#endif
+
+#ifndef NM_802_11_AP_SEC_PAIR_WEP104
+#  define NM_802_11_AP_SEC_PAIR_WEP104 0x00000001
+#endif
+
+namespace {
+Logger logger(LOG_LINUX, "LinuxNetworkWatcher");
+}
+
+static inline bool checkSecurityFlags(int securityFlags) {
```

Comment:
It may well be my intexperience with bitwise operation, or just my inexperience in general, but I find a method like this very hard to parse. It may be that expanding it out to make it more verbose is unacceptably inefficient. If so, some tests that spell out the various tests that are being tested would be great.